pkg/apis/kubeflow.org/v1/jax_types.go

```diff
@@ -0,0 +1,104 @@
+// Copyright 2024 The Kubeflow Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package v1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+const (
+	// JAXJobDefaultPortName is name of the port used to communicate between Coordinator and Workers.
+	JAXJobDefaultPortName = "jaxjob-port"
+	// JAXJobDefaultContainerName is the name of the JAXJob container.
+	JAXJobDefaultContainerName = "jax"
+	// JAXJobDefaultPort is default value of the port.
+	JAXJobDefaultPort = 6666
+	// JAXJobDefaultRestartPolicy is default RestartPolicy for JAXReplicaSpecs.
+	JAXJobDefaultRestartPolicy = RestartPolicyNever
+	// JAXJobKind is the kind name.
+	JAXJobKind = "JAXJob"
+	// JAXJobPlural is the JAXJobPlural for JAXJob.
+	JAXJobPlural = "jaxjobs"
+	// JAXJobSingular is the singular for JAXJob.
+	JAXJobSingular = "jaxjob"
+	// JAXJobFrameworkName is the name of the ML Framework
+	JAXJobFrameworkName = "jax"
+	// JAXJobReplicaTypeCoordinator is the type of Coordinator of distributed JAX
+	JAXJobReplicaTypeCoordinator ReplicaType = "Coordinator"
```

Comment:
As we discussed today, we don't need to have separate replica for JAX, since Pod template will be the same and Worker with index 0 will the Coordinator.
Does it sound good to you @tenzen-y ?

Reply:
I'm ok with v1 API since we can not apply separate runPolicies only to coordinator.
In the v2 API, I believe that dedicated replica for the coordinator would be worth it since we often want to treat the coordinator runPolicies.

But, v2 API is a separate discussion.

Reply:
Makes sense @tenzen-y. Do you think, setting various runPolicy for JAX Coordinator and Workers makes sense even that JAX follows: SPMD model: https://jax.readthedocs.io/en/latest/multi_process.html#launching-jax-processes ?
E.g. all JAX hosts should be always running.

Reply:
In the v1 Training API, we don't have any ability to configure runPolicy on each replicas (coordinator and worker).
So, let's discuss if we should add the Coordinator role in the v2 API.